### PR TITLE
[BACKLOG-36954] - Changed dialogs to responsive and removed fixed lengths

### PR DESCRIPTION
**Pull request by jsaraf**

### CODE REVIEW

**Comment by jsaraf** [comment]
Please review @pentaho/wcag

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 52s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 59s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by dcleao** [comment]
Replaces https://github.com/pentaho/pentaho-commons-gwt-modules/pull/875.

**Comment by jsaraf** [comment]
> Replaces #875.

Yes, pulled in latest code and made appropriate changes as required.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 1]
```diff
@@ -573,15 +571,21 @@ div#promptPanel div.prompt-panel {
 }
 
 div#toppanel {
-  margin-top: 20px;
+  margin-top: unset;
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:574]: Report Prompts are out-of-scope, AFAIK.
Also, these changes would affect prompts even when used outside the Parameter UI.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 2]
```diff
@@ -573,15 +571,21 @@ div#promptPanel div.prompt-panel {
...
 
 div#promptPanel div.parameter {
-  margin-bottom: 20px;
+  margin-bottom: unset;
+}
+
+.parameter-container > tbody > tr > td > div > table > tbody {
+  gap: var(--layout-gap-v) var(--layout-gap-h);
+  display: flex;
+  flex-direction: column;
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:588]: Idem, for Prompts to be out-of-scope.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 3]
```diff
@@ -1307,7 +1279,7 @@ For IE 9 and IE10, need to add padding to the bottom of some dialogs generated w
 }
 
 .pentaho-dialog .button-panel .pentaho-button {
-  margin-left: 10px;
+  margin-left: 10px !important;
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:1282]: This is not needed, AFAICT. Also, we should not be changing the non-responsive base dialog styles.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 4]
```diff
@@ -9325,3 +9393,64 @@ body.bootstrap .modal.confirmationDialog .dialog div.body {
...
+  margin-bottom: 5px;
+}
+
+.schedulesPanel .pentaho-table th {
+  font-size: 12px;
+  border-spacing: 0;
+  color: #FFF;
+  padding: 8px 10px 10px 10px;
+  font-weight: normal;
+}
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:9411]: These two rules belong to the Schedules perspective, which is loaded inside Mantle. Should be moved back to `mantleRuby.css`.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 5]
```diff
@@ -9325,3 +9393,64 @@ body.bootstrap .modal.confirmationDialog .dialog div.body {
...
+
+.schedule-caption-panel {
+  margin-top: unset;
+}
+
+.schedule-dialog-button-panel {
+  padding-right: 3px;
+  padding-bottom: 2px;
+  width: 100%;
+}
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:9456]: Should be moved to the above group `/* region Schedule Wizard Dialogs */`, and each rule placed under the appropriate sub-section, if any.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 6]
```diff
@@ -1682,6 +1679,76 @@ button#schedule-dialog-select-button {
...
+  /* This will fail when --frame-content-height is not set,
+     falling back to previous declaration with auto.
+   */
+  height: calc(var(--frame-content-height, auto) - 2 * var(--focus-ring-padding)) !important;
+}
+
+/* move to anaRuby.css */
+:where(.pentaho-ana.responsive).parameter-ui #standardDialog {
+  width: 100% !important;
+}
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:1748]: Should be moved to `anaRuby.css`.

**Review by dcleao — CHANGES_REQUESTED** [review]

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/wizards/AbstractWizardDialog.java
[the remark this answers is not in the document]
Comment (jsaraf) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/wizards/AbstractWizardDialog.java:1]: Adressing in the next commit.

File: widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/ScheduleOutputLocationDialog.java [hunk 7]
```diff
@@ -327,4 +337,4 @@ public void setOkButtonText( String text ) {
   public void setScheduleNameText( String text ) {
     scheduleNameLabel.setText( text );
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (jsaraf) [reply widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/ScheduleOutputLocationDialog.java:340]: Reverted.

File: widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/ScheduleParamsDialog.java [hunk 8]
```diff
@@ -127,8 +131,8 @@ public void initDialog() {
     setParametersUrl( EnvironmentHelper.getFullyQualifiedURL() + "api/repos/" + urlPath + "/parameterUi" + urlParams ); //$NON-NLS-1$ //$NON-NLS-2$
     wizardDeckPanel.setHeight( "100%" ); //$NON-NLS-1$
 
-    wizardDeckPanel.getElement().getParentElement().addClassName( "schedule-dialog-content" );
-    wizardDeckPanel.getElement().getParentElement().removeClassName( "dialog-content" );
+    setSize( "650px", "450px" );
```
[the remark this answers is not in the document]
Comment (jsaraf) [reply widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/ScheduleParamsDialog.java:134]: Removed the duplicate.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 1 above)
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:574]: Reverted the change.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 2 above)
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:588]: Removed.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 3 above)
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:1282]: Reverted.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 4 above)
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:9411]: Moved back.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 6 above)
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:1748]: Moved.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 54s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 7s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/11/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — CHANGES_REQUESTED** [review]
Besides the below comments, please check the sonar code smells. Also, please do not include merge commits in your proposed commits (instead squash/rebase on top of wcag).

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 9]
```diff
@@ -4723,6 +4825,7 @@ div.dijitDateTextBox.dijitComboBox {
   flex-direction: column;
   flex: none;
 }
+/* endregion .folder, .folders ... */
```
[the remark this answers is not in the document]
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:4828]: Most probably. Will be removing the line.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 10]
```diff
@@ -9325,3 +9428,5 @@ body.bootstrap .modal.confirmationDialog .dialog div.body {
     text-overflow: ellipsis;
   }
 }
+
+/* endregion Main Toolbar */
```
[the remark this answers is not in the document]
Comment (jsaraf) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:9432]: Same as above.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 55s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/13/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 55s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **398**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-877/14/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by hv-vamsi — APPROVED** [review]

**Pull request merged**